src/read.c

```diff
@@ -4268,24 +4272,29 @@ static avifResult avifMetaFindAlphaItem(avifMeta * meta,
     if (alphaItemCount != colorItemCount) {
         // Not all the color items had an alpha auxiliary attached to it. Report this case as an image without alpha channel.
         avifFree(alphaItemIndices);
-        *alphaItem = NULL;
+        *alphaItemIndex = -1;
         *isAlphaItemInInput = AVIF_FALSE;
         return AVIF_RESULT_OK;
     }
-    const avifResult result = avifMetaFindOrCreateItem(meta, maxItemID + 1, alphaItem); // Create new empty item.
+    avifDecoderItem * alphaItem;
+    const avifResult result = avifMetaFindOrCreateItem(meta, maxItemID + 1, &alphaItem); // Create new empty item.
     if (result != AVIF_RESULT_OK) {
         avifFree(alphaItemIndices);
         *isAlphaItemInInput = AVIF_FALSE;
         return result;
     }
-    memcpy((*alphaItem)->type, "grid", 4); // Make it a grid and register alpha items as its tiles.
-    (*alphaItem)->width = colorItem->width;
-    (*alphaItem)->height = colorItem->height;
+    // avifMetaFindOrCreateItem() could invalidate all existing item pointers. So repopulate the colorItem pointer.
```

Comment:
Nit: "repopulate" implies filling out the members of a struct. Perhaps "reassign" or "reset"?

Reply:
Done.